connectors/src/connectors/github/temporal/workflows.ts

```diff
@@ -144,87 +155,128 @@ export async function githubRepoSyncWorkflow(
   githubInstallationId: string,
   repoName: string,
   repoId: number,
-  repoLogin: string
+  repoLogin: string,
+  connectorId: string,
+  syncCodeOnly: boolean
 ) {
   const loggerArgs = {
     dataSourceName: dataSourceConfig.dataSourceName,
     workspaceId: dataSourceConfig.workspaceId,
     githubInstallationId,
     repoName,
     repoLogin,
+    syncCodeOnly: syncCodeOnly ? "true" : "false",
   };
 
-  const queue = new PQueue({
-    concurrency: MAX_CONCURRENT_ISSUE_SYNC_ACTIVITIES_PER_WORKFLOW,
-  });
-  const promises: Promise<void>[] = [];
-
-  let pageNumber = 1; // 1-indexed
-  for (;;) {
-    const resultsPage = await githubGetRepoIssuesResultPageActivity(
-      githubInstallationId,
-      repoName,
-      repoLogin,
-      pageNumber,
-      loggerArgs
-    );
-    if (!resultsPage.length) {
-      break;
-    }
-    pageNumber += 1;
-    for (const issueNumber of resultsPage) {
-      promises.push(
-        queue.add(() =>
-          githubUpsertIssueActivity(
-            githubInstallationId,
-            repoName,
-            repoId,
-            repoLogin,
-            issueNumber,
-            dataSourceConfig,
-            loggerArgs,
-            true // isBatchSync
-          )
-        )
-      );
-    }
-  }
+  if (!syncCodeOnly) {
+    const queue = new PQueue({
+      concurrency: MAX_CONCURRENT_ISSUE_SYNC_ACTIVITIES_PER_WORKFLOW,
+    });
+    const promises: Promise<void>[] = [];
 
-  let nextCursor: string | null = null;
-  for (;;) {
-    const { cursor, discussionNumbers } =
-      await githubGetRepoDiscussionsResultPageActivity(
+    let pageNumber = 1; // 1-indexed
+    for (;;) {
+      const resultsPage = await githubGetRepoIssuesResultPageActivity(
         githubInstallationId,
         repoName,
         repoLogin,
-        nextCursor,
+        pageNumber,
         loggerArgs
       );
-    for (const discussionNumber of discussionNumbers) {
-      promises.push(
-        queue.add(() =>
-          githubUpsertDiscussionActivity(
-            githubInstallationId,
-            repoName,
-            repoId,
-            repoLogin,
-            discussionNumber,
-            dataSourceConfig,
-            loggerArgs,
-            true // isBatchSync
+      if (!resultsPage.length) {
+        break;
+      }
+      pageNumber += 1;
+      for (const issueNumber of resultsPage) {
+        promises.push(
+          queue.add(() =>
+            githubUpsertIssueActivity(
```

Comment:
I guess this one should also have been an object params

Reply:
will change :+1:

Reply:
Yes 😅  I wrote this one

Reply:
I'll make a pass to cleanup all others 👍